src/hotspot/share/gc/parallel/mutableSpace.hpp

```diff
@@ -25,35 +25,40 @@
 #ifndef SHARE_GC_PARALLEL_MUTABLESPACE_HPP
 #define SHARE_GC_PARALLEL_MUTABLESPACE_HPP
 
-#include "gc/parallel/immutableSpace.hpp"
+#include "memory/allocation.hpp"
+#include "memory/iterator.hpp"
 #include "memory/memRegion.hpp"
 #include "utilities/copy.hpp"
+#include "utilities/globalDefinitions.hpp"
+#include "utilities/macros.hpp"
 
 class WorkGang;
 
-// A MutableSpace is a subtype of ImmutableSpace that supports the
-// concept of allocation. This includes the concepts that a space may
-// be only partially full, and the query methods that go with such
-// an assumption. MutableSpace is also responsible for minimizing the
+// A MutableSpace supports the concept of allocation. This includes the
+// concepts that a space may be only partially full, and the query methods
+// that go with such an assumption.
+//
+// MutableSpace is also responsible for minimizing the
 // page allocation time by having the memory pretouched (with
 // AlwaysPretouch) and for optimizing page placement on NUMA systems
 // by make the underlying region interleaved (with UseNUMA).
 //
-// Invariant: (ImmutableSpace +) bottom() <= top() <= end()
+// Invariant: bottom() <= top() <= end()
 // top() is inclusive and end() is exclusive.
```

Comment:
If end() is exclusive then shouldn't the invariant be `< end()`?

Reply:
I also think that top() is also exclusive as in other collectors.

@dholmes-ora : e.g. bottom == top == end means the space is empty. These two lines are not disagreeing with each other.